test/services/prompter/database-prompts.js

```diff
@@ -0,0 +1,768 @@
+    });
+  });
+
+  describe('Handling connection url : ', () => {
+    describe('When the dbConnectionUrl option is requested', () => {
+      describe('and the dbConnectionUrl has already been passed in', () => {
+        describe('and the dbConnectionUrl is valid', () => {
+          before(() => {
+            requests.push('dbConnectionUrl');
+          });
+
+          after(() => {
+            resetParams();
+          });
+
+          it('should add the dbConnectionUrl to the configuration', async () => {
+            const databasePrompts = new DatabasePrompts(program, envConfig, prompts, requests);
+
+            program.connectionUrl = 'postgres://username:password@host:port/database';
+
+            await databasePrompts.handleConnectionUrl();
+
+            expect(envConfig.dbConnectionUrl).to.equal(program.connectionUrl);
+          });
+
+          it('should add the dbDialect to configuration', async () => {
+            const databasePrompts = new DatabasePrompts(program, envConfig, prompts, requests);
+
+            program.connectionUrl = 'postgres://username:password@host:port/database';
+
+            await databasePrompts.handleConnectionUrl();
+
+            expect(envConfig.dbDialect).to.equal('postgres');
+          });
+
+          it('should add the mongo dbDialect to configuration when using mongo+srv', async () => {
+            const databasePrompts = new DatabasePrompts(program, envConfig, prompts, requests);
+
+            program.connectionUrl = 'mongodb+srv://username:password@host1:port1/database';
+
+            await databasePrompts.handleConnectionUrl();
+
+            expect(envConfig.dbDialect).to.equal('mongodb');
+          });
+        });
+
+        describe('and the dbConnectionUrl is invalid', async () => {
+          let loggerSpy;
+          let eventSenderSpy;
+          let databasePrompts;
+          let processStub;
+
+          before(async () => {
+            processStub = sinon.stub(process, 'exit');
+            loggerSpy = sinon.spy(logger, 'error');
+            eventSenderSpy = sinon.spy(eventSender, 'notifyError');
+            requests.push('dbConnectionUrl');
+            program.connectionUrl = 'invalid';
+            databasePrompts = new DatabasePrompts(program, envConfig, prompts, requests);
+            await databasePrompts.handleConnectionUrl();
+          });
+
+          after(() => {
+            resetParams();
+            loggerSpy.restore();
+            eventSenderSpy.restore();
+            processStub.restore();
+          });
+
+          it('should terminate the process with exit code 1', async () => {
+            expect(processStub.calledOnce).to.equal(true);
+            expect(processStub.calledWith(1)).to.equal(true);
+          });
+
+          it('should log an error message', async () => {
+            const message = 'Cannot parse the database dialect. Please, check the syntax of the database connection string.';
```

Comment:
[Misc] I think this should come from the new `utils/messages.js` to be easier to maintain as you will just have to rename the message in one place ^^